elastic/kibana
# [Security solution] Guided onboarding, alerts & cases design updates

## Summary

Makes updates to the guided onboarding design for the alerts and cases tour [as requested here](https://github.com/elastic/kibana/pull/143598#issuecomment-1290482225).

1. Adds a "Next" button to all tour steps. Adds an `onClick` property to `SecurityTourStep` to accomplish this. When next is click, either `incrementStep` if no `onClick` provided or call the `onClick`. Therefore, any provided `onClick` should handle incrementing the tour step
2. Focuses on Next buttons throughout tour (removes `hideNextButton`)
3. Pre-populates a sample case
4. Next button on create case step implements to a new step that scrolls down and calls attention to the create case button
5. Implements a new last step that has user click through to case details, where the case tour ends
6. Fixes bug with expanded rule name cell 
7. Fix a few places where the step `number` was still being used instead of `AlertsCasesTourSteps`

Here is what all of that looks like:

https://user-images.githubusercontent.com/6935300/200897274-ca027920-a358-4857-ab71-5b7c774fa8c1.mov




## To test
1. add feature flag to kibana.dev.yml: `xpack.securitySolution.enableExperimental: ['guidedOnboarding']`
2. add cloud test feature flag: `xpack.cloud.id: 'test'`
3. Run kibana with examples: `yarn start --run-examples`
4. Ensure you have alerts
5. Go to `/app/guidedOnboardingExample`
6. Use the "Set guide state" section to set Guide to `security` and Step ID to `alertsCases`
9. The Setup guide menu will automatically open to Security step 3. Hit the "Start" button
10. You will be redirected to the alerts page and at this point, we will start on the Alerts and cases tour. Click through the steps to complete the tour and check the fixes mentioned above

## Changes to the `cases` app

For the cases tour step, our designer requested that when the create case flyout opens, the form is pre-populated.

- To achieve the pre-populated form, I added a new property to the `CreateCaseFlyout` of `initialValue: Partial<Case>`. This value is piped to the form context and spread with in the definition of the create form: `defaultValue: { ...initialCaseValue, ...initialValue }`
- Added `tour-step` properties to Eui elements to use as references for anchors and `focus()` instead of `data-test-subj`
